applications/luci-app-tailscale-community/htdocs/luci-static/resources/view/tailscale.js

```diff
@@ -54,21 +69,102 @@ function getRunningStatus() {
 }
 
 function formatBytes(bytes) {
-    let bytes_num = parseInt(bytes, 10);
+    const bytes_num = parseInt(bytes, 10);
     if (isNaN(bytes_num) || bytes_num === 0) return '-';
-    let k = 1024;
-    let sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
-    let i = Math.floor(Math.log(bytes_num) / Math.log(k));
+    const k = 1000;
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+    const i = Math.floor(Math.log(bytes_num) / Math.log(k));
     return parseFloat((bytes_num / Math.pow(k, i)).toFixed(2)) + ' ' + sizes[i];
 }
 
+function formatLastSeen(dateString) {
+    if (!dateString) return 'N/A';
+    if (dateString === '0001-01-01T00:00:00Z') {
+        return 'Now';
+    }
+    const lastSeenDate = new Date(dateString);
+    // Check for a valid date.
+    if (isNaN(lastSeenDate.getTime())) {
+        return 'Invalid Date';
+    }
+    const now = new Date();
+    const diffSeconds = Math.round((now - lastSeenDate) / 1000);
+    if (diffSeconds < 0) {
+        return lastSeenDate.toLocaleString();
+    }
+    if (diffSeconds < 60) {
+        return 'Just now';
+    }
+    const diffMinutes = Math.floor(diffSeconds / 60);
+    if (diffMinutes < 60) {
+        return `${diffMinutes} ${diffMinutes === 1 ? 'minute' : 'minutes'} ago`;
+    }
+    const diffHours = Math.floor(diffMinutes / 60);
+    if (diffHours < 24) {
+        return `${diffHours} ${diffHours === 1 ? 'hour' : 'hours'} ago`;
+    }
+    const diffDays = Math.floor(diffHours / 24);
+    if (diffDays < 30) {
+        return `${diffDays} ${diffDays === 1 ? 'day' : 'days'} ago`;
+    }
+    const year = lastSeenDate.getFullYear();
+    const month = String(lastSeenDate.getMonth() + 1).padStart(2, '0');
+    const day = String(lastSeenDate.getDate()).padStart(2, '0');
+    return `${year}-${month}-${day}`;
+}

```

Comment:
Not 100% sure about the last line `toISOString()` which might need adjusting, but try this function: 
```js
function formatLastSeen(d) {
	if (!d) return 'N/A';
	if (d === '0001-01-01T00:00:00Z') return 'Now';
	const t = new Date(d);
	if (isNaN(t)) return 'Invalid Date';
	const diff = (Date.now() - t) / 1000;
	if (diff < 0) return t.toLocaleString();
	if (diff < 60) return 'Just now';
	const mins = diff / 60, hrs = mins / 60, days = hrs / 24;
	const fmt = (n, u) => `${Math.floor(n)} ${Math.floor(n) === 1 ? u : u + 's'} ago`;
	if (mins < 60) return fmt(mins, 'minute');
	if (hrs < 24) return fmt(hrs, 'hour');
	if (days < 30) return fmt(days, 'day');
	return t.toISOString().slice(0, 10);
}

```